core/src/main/java/io/questdb/griffin/engine/functions/finance/WeightedMidPriceFunctionFactory.java

```diff
@@ -0,0 +1,105 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.finance;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.DoubleFunction;
+import io.questdb.griffin.engine.functions.QuaternaryFunction;
+import io.questdb.std.IntList;
+import io.questdb.std.Numbers;
+import io.questdb.std.ObjList;
+
+public class WeightedMidPriceFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "wmid(DDDD)";
+    }
+
+    @Override
+    public Function newInstance(int position, ObjList<Function> args, IntList argPositions,
+                                CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
+        return new WeightedMidPriceFunction(args.getQuick(0), args.getQuick(1), args.getQuick(2), args.getQuick(3));
```

Comment:
The arguments order isn't consistent. Currently it's `bidSize, bid, ask, askSize`. It should be either `bid, bidSize, ask, askSize` or `bidSize, bid, askSize, ask`. WDYT? Sorry for missing this before.

Reply:
`bidSize, bid, ask, askSize` follows the same order commonly displayed on trading systems like Bloomberg, Fidessa or Triton.

Reply:
Ok, so it's kept this way on purpose. Thanks for the clarification. Maybe it's worth adding a clarifying comment, so that no one gets tempted to change the order in the future?